### ISSUE TITLE
Badge request : openuserjs.org, greasyfork.org

### ISSUE THREAD

**Issue by NatoBoram** [badges/shields#5505]
:clipboard: **Description**

Services : openuserjs.org, greasyfork.org

Information : Install count, rating, version code

Examples : 

![OpenUserJS](https://img.shields.io/badge/OpenUserJS-3%20installs-%232c3e50)
![GreasyFork](https://img.shields.io/badge/GreasyFork-3%20installs-%23990000)

:link: **Data**

* No public API
* No key needed

Data can be directly accessed from the HTML file.

:microphone: **Motivation**

This feature can help assert a certain "reputation" based on install count for public scripts.

More specifically, I want to be able to show off that people have installed my userscript.

**Comment by calebcartwright** [comment]
Thanks for the request @NatoBoram. However, we can't do html scraping to retrieve the requisite data for the badges. If these tools/services do not provide an endpoint to retrieve the data, then we can't provide native badges for them, but you could leverage our [Endpoint badge](https://shields.io/endpoint) instead.

**Comment by NatoBoram** [comment]
Found the API for GreasyFork!

* `https://greasyfork.org/scripts/411035.json`
* ![GreasyFork](https://img.shields.io/badge/dynamic/json?color=%23990000&label=GreasyFork&query=total_installs&suffix=%20installs&url=https%3A%2F%2Fgreasyfork.org%2Fscripts%2F411035.json)
* `https://img.shields.io/badge/dynamic/json?color=%23990000&label=GreasyFork&query=total_installs&suffix=%20installs&url=https%3A%2F%2Fgreasyfork.org%2Fscripts%2F411035.json`

**Comment by PyvesB** [comment]
@NatoBoram is the API documented anywhere? It would be quite useful!

**Comment by jerone** [comment]
Related issue: #829

**Comment by NatoBoram** [comment]
I don't think there's a documentation anywhere. I found out in https://github.com/JasonBarnabe/greasyfork/issues/247 that you can add `.json` to some web pages and it'll return a JSON response. 

Their example is https://greasyfork.org/users/173559.json, so I tried on my script and https://greasyfork.org/scripts/411035.json works.

I asked in https://github.com/JasonBarnabe/greasyfork/issues/787.

**Comment by Martii** [comment]
Well I'll have to reread Shields documents and find out how they are actually querying any site to add this to OUJS e.g. this part `query=total_installs&suffix=%20installs`... if it's a QSP it may be doable on our own since we do have a documented .json routine *(just no QSP's atm to pull individual elements out)*... if it's a headered request that will need to be a collaboration with Shields and some tighter restrictions. 

In other words need to dig in their docs and code to see for sure.

**Comment by NatoBoram** [comment]
Personally, I just filled the _Dynamic_ form with those values

* `GreasyFork`
* `https://greasyfork.org/scripts/411035.json`
* `total_installs`
* `#990000`
*
* ` installs`

![image](https://user-images.githubusercontent.com/10495562/92668727-a2213180-f2dd-11ea-95af-6761d8ce6388.png)

The `query` is just a [JSON Path](https://jsonpath.com/). The only variable here is `411035`, the script ID.

**Comment by Martii** [comment]
@NatoBoram 
Well the documentation is at https://openuserjs.org/about/Frequently-Asked-Questions#q-does-openuserjs-org-have-meta- and my Unit Test for the site is at https://openuserjs.org/scripts/Marti/oujs_-_Meta_View *(which just happens to utilize install count for that .user.js to fill out the phantom URL UI)* which will show the .json fields we collect. If you need something added let us know on the issue you created on OUJS GH dev. If it's not too DB intensive it can probably be added.

So if Shields can parse that on their end via their QSP's it may work for you as is. Make note some of the info you are looking for is not in the `UserScript` block property but in the `OpenUserJS` block. i.e. we have multiple tiers in the JSON. I specifically made the .json available myself and it was adopted later by others.

@calebcartwright 

> we can't do html scraping to retrieve the requisite data for the badges.

We appreciate the not scraping aspect. :smile_cat:

**Comment by NatoCMI** [comment]
Thanks @Martii! I got it working.

* `json`
* `OpenUserJS`
* `https://openuserjs.org/meta/NatoBoram/YouTube_Comment_Blacklist.meta.json`
* `$.OpenUserJS.installs[0].value`
* `#2c3e50`
*
* ` installs`

[![OpenUserJS](https://img.shields.io/badge/dynamic/json?color=%232c3e50&label=OpenUserJS&query=%24.OpenUserJS.installs%5B0%5D.value&suffix=%20installs&url=https%3A%2F%2Fopenuserjs.org%2Fmeta%2FNatoBoram%2FYouTube_Comment_Blacklist.meta.json)](https://openuserjs.org/scripts/NatoBoram/YouTube_Comment_Blacklist)

I do wonder why `installs` is an array, though.

**Comment by NatoBoram** [comment]
https://github.com/JasonBarnabe/greasyfork/wiki/API

**Comment by KaKi87** [comment]
Hello, any news on this ? Thanks !

**Comment by calebcartwright** [comment]
> Hello, any news on this ? Thanks !

We post updates as and when they are available, so as a general rule of thumb, the absence of updates should be interpreted as meaning there are no updates and there is no news. That applies here as well.

A json endpoint is a perfectly viable data retrieval target for us, and we've a number of badges that fetch data via a similar mechanism. As such, it should indeed be possible for someone to submit a PR with the requisite code to add these badges. We've got a solid [tutorial](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md) that provides a step by step guide for anyone sufficiently interested in working on this.

We view Shields as a community project that's stewarded by us as a small maintainer team (with limited bandwidth). We're happy to provide guidance and assistance to anyone from the community that wants to try to implement the badge(s), but to be fully transparent, it's rather unlikely that it'll ever be implemented by a maintainer.

In the interim, a number of folks have posted available options which utilize our Dynamic Badge feature that can be leveraged today.
